### PR TITLE
build: bundle uncrustify

### PR DESCRIPTION
**Pull request by dundargoc**
Uncrustify is sensitive to version changes, which causes friction for
contributors that doesn't have that exact version. It's also simpler to
download and install the correct version than to have bespoke version
checking.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
       - name: Install dependencies
         run: |
           ./.github/scripts/install_deps.sh
-          brew install stylua uncrustify
+          brew install stylua
```
Comment (justinmk) [inline .github/workflows/test.yml:41]: brew can be really slow in CI, so it might even worth fetching their binary instead of using `brew` in CI.

Comment (dundargoc) [reply .github/workflows/test.yml:41]: Good call. The lint times are getting gross so will need to make a more thorough refactor of this, one of which is to dump brew 👍

Comment (dundargoc) [reply .github/workflows/test.yml:41]: Follow-up PR though

Comment (carlocab) [reply .github/workflows/test.yml:41]: If you do
```sh
export HOMEBREW_NO_AUTO_UPDATE=1
export HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK=1
export HOMEBREW_NO_INSTALL_UPGRADE=1
export HOMEBREW_NO_INSTALL_CLEANUP=1
brew untap homebrew/core
```
this should substantially speed up your usage of `brew`.

Comment (dundargoc) [reply .github/workflows/test.yml:41]: Didn't make a difference.

**Pull request merged**